ngx_http_cookie_flag_filter_module.c

```diff
@@ -235,57 +235,27 @@ ngx_http_cookie_flag_filter_init(ngx_conf_t *cf)
-        if (tmp.data == NULL) {
-            return NGX_ERROR;
-        }
-        tmp.len = ngx_sprintf(tmp.data, "%V; HttpOnly", &header->value) - tmp.data;
-        header->value.data = tmp.data;
-        header->value.len = tmp.len;
-    }
-
-    if (cookie->secure == 1 && ngx_strcasestrn(header->value.data, "; secure", 8 - 1) == NULL) {
-        tmp.data = ngx_pnalloc(r->pool, header->value.len + sizeof("; secure") - 1);
-        if (tmp.data == NULL) {
-            return NGX_ERROR;
-        }
-        tmp.len = ngx_sprintf(tmp.data, "%V; secure", &header->value) - tmp.data;
-        header->value.data = tmp.data;
-        header->value.len = tmp.len;
-    }
-
-    if (cookie->samesite == 1 && ngx_strcasestrn(header->value.data, "; SameSite", 10 - 1) == NULL) {
-        tmp.data = ngx_pnalloc(r->pool, header->value.len + sizeof("; SameSite") - 1);
-        if (tmp.data == NULL) {
-            return NGX_ERROR;
-        }
-        tmp.len = ngx_sprintf(tmp.data, "%V; SameSite", &header->value) - tmp.data;
-        header->value.data = tmp.data;
-        header->value.len = tmp.len;
-    }
-
-    if (cookie->samesite_lax == 1 && ngx_strcasestrn(header->value.data, "; SameSite=Lax", 14 - 1) == NULL) {
-        tmp.data = ngx_pnalloc(r->pool, header->value.len + sizeof("; SameSite=Lax") - 1);
-        if (tmp.data == NULL) {
-            return NGX_ERROR;
-        }
-        tmp.len = ngx_sprintf(tmp.data, "%V; SameSite=Lax", &header->value) - tmp.data;
-        header->value.data = tmp.data;
-        header->value.len = tmp.len;
-    }
-
-    if (cookie->samesite_strict == 1 && ngx_strcasestrn(header->value.data, "; SameSite=Strict", 17 - 1) == NULL) {
-        tmp.data = ngx_pnalloc(r->pool, header->value.len + sizeof("; SameSite=Strict") - 1);
-        if (tmp.data == NULL) {
-            return NGX_ERROR;
-        }
-        tmp.len = ngx_sprintf(tmp.data, "%V; SameSite=Strict", &header->value) - tmp.data;
-        header->value.data = tmp.data;
-        header->value.len = tmp.len;
-    }
+    u_char *p, *last;
+
+#define NGX_CHECK_APPEND_COOKIE_FLAG(a, b) do {                     \
+  last = header->value.data + header->value.len;                    \
+  if (cookie->a == 1 && ngx_strlcasestrn(header->value.data, last,  \
+        (u_char *)b, sizeof(b) - 1) == NULL) {                      \
+    p = ngx_pnalloc(r->pool, header->value.len + sizeof(b));        \
+    if (p == NULL) {                                                \
+      return NGX_ERROR;                                             \
+    }                                                               \
+    header->value.len = ngx_sprintf(p, "%V" b, &header->value) - p; \
+    header->value.data = p;                                         \
+    p[header->value.len] = '\0';                                    \
+  }                                                                 \
+} while(0)
```

Comment:
What is the purpose of the 'while' loop here, the original code has no such loop?